lib/qbo_api.rb

```diff
@@ -112,26 +112,42 @@ def request(method, path:, entity: nil, payload: nil, params: nil)
   end
 
   def response(resp, entity: nil)
-    data = JSON.parse(resp.body)
-    if entity
-      entity_response(data, entity)
-    else
-      data
-    end
+    data = parse_response_body(resp)
+    entity ? entity_response(data, entity) : data
   rescue => e
-    # Catch fetch key errors and just return JSON
+    # NOTE(BF): Rescue all StandardError and return parsed JSON body
```

Comment:
You can take out all of the comments in this PR. I think the code is readable enough.

Reply:
Will do

(Added the comment because I don't think we should be rescuing all standard error here but don't want to break existing functionality).